src/engine/sparqlExpressions/StringExpressions.cpp

```diff
@@ -399,6 +399,28 @@ class ConcatExpression : public detail::VariadicExpression {
 using EncodeForUriExpression =
     StringExpressionImpl<1, decltype(encodeForUriImpl)>;
 
+// TO_NUMERIC (int or double)
+auto InvalidArg = std::errc::invalid_argument;
+auto OutOfRange = std::errc::result_out_of_range;
+
+template <typename T>
+[[maybe_used]] auto toNumeric = [] (std::optional<std::string> input) {
+  if (!input.has_value()) {
+    return Id::makeUndefined();
+  } else {
+    T res{};
+    auto str = boost::lexical_cast<std::string>(*input);
+    auto conv = std::from_chars(str.data(), str.data() + str.size(), res);
+    if (conv.ec == InvalidArg || conv.ec == OutOfRange) {
+      return Id::makeUndefined();
+    }
+    if (std::is_same_v<T, double>) { return Id::makeFromDouble(res); }
```

Comment:
```suggestion
    if constexpr (std::is_same_v<T, double>) { return Id::makeFromDouble(res); }
```
(The compiler will optimize this anyway, but makes the code clearer and you learn another feature).